plugins/quetz_sql_authenticator/quetz_sql_authenticator/api.py

```diff
@@ -72,8 +73,16 @@ def _create(
         username=username, password_hash=_calculate_hash(password)
     )
     db.add(credentials)
-    db.commit()
-
+    try:
+        db.commit()
+    except IntegrityError:
+        raise HTTPException(
+            status_code=status.HTTP_409_CONFLICT,
+            detail=f"""
+            A database error occured.
+            This is most likely due to the user {username} already existing.
+            """,
+        )
```

Comment:
can we make this error message a bit more detailed?

i suggest the following:

1. Parse the integrity error
2. Figure out if the integrity error is due to a name conflict
3. If yes, return a corresponding error message (something like 'username already exists.')
4. If no, return the integrity error message (something like: 'a database error occurred: {integrity error message}')